File: front/src/app/core/model/dto/deck-dto.ts
import { IndexedCardDetailDTO } from './card-detail-dto';
import { IndexedCardImageDTO } from './card-image-dto';

export class DeckDTO {
  id: number;
  name: string;
  images: Array<IndexedCardImageDTO>;
  mainDeck: Array<IndexedCardDetailDTO>;
  extraDeck: Array<IndexedCardDetailDTO>;
  sideDeck: Array<IndexedCardDetailDTO>;

  constructor(deckDTO: DeckDTO) {
    this.id = deckDTO.id;
    this.name = deckDTO.name;
    this.images = deckDTO.images;
    this.mainDeck = deckDTO.mainDeck;
    this.extraDeck = deckDTO.extraDeck;
    this.sideDeck = deckDTO.sideDeck;
  }
}

File: back/src/main/java/com/skytrix/model/dto/deck/DeckDTO.java
package com.skytrix.model.dto.deck;

import java.util.List;

import lombok.Data;

@Data
public class DeckDTO {
    private Long id;
    private String name;
    private List<IndexedCardImageDTO> images;
    private List<IndexedCardDetailDTO> mainDeck;
    private List<IndexedCardDetailDTO> extraDeck;
    private List<IndexedCardDetailDTO> sideDeck;
}
